How-to-upgrade-review-version.md

```diff
@@ -0,0 +1,399 @@
+* @types/shelljs
+* @types/eslint-scope
+
+If you are really, truly, completely in stealth mode, make sure not to
+save these installations in package.json. That's a bit extreme,
+though, isn't it?
+
+Anyway, these three types packages still didn't work:
+
+* @types/shelljs
+* @types/estree
+* @types/eslint-scope
+
+They fail for different reasons, though. shelljs and es-lint-scope are
+just missing whole chunks of the package's types. estree *has* all the
+correct types, but the types aren't imported correctly. So now you can
+decide where to start fixing errors.
+
+You can
+
+1. Add missing types in dependencies.
+2. Fix references to types in dependencies.
+4. Add missing types in your own code.
+3. Work around missing types in dependencies.
+5. Fix errors in existing types.
+6. Add type annotations to everything else.
+
+## Add missing types in dependencies
+
+Let's start with @types/shelljs. In Makefile.js, I see a few errors.
+The first is that the module `require('shelljs/make')` isn't found.
+The second group of errors is that the names 'find', 'echo' and a few
+others aren't found.
+
+These errors are related. It turns out that shelljs doesn't even
+include shelljs/make right now. It's completely missing. I looked it
+up and shelljs/make does two things:
+
+1. Add a global object named 'target' that allows you to add make
+targets.
+2. Add the contents of the parent shelljs to the global scope.
+
+So I want to add this to Definitely Typed. My first step is to create
+make.d.ts inside node_modules/@types/shelljs/. This is just for
+development, because it makes it super easy to test that I'm actually
+fixing the missing stuff.
+
+I start with this:
+
+```ts
+import * as shelljs from './';
+declare global {
+  const cd: typeof shelljs.cd;
+  const pwd: typeof shelljs.pwd;
+  // ... all the rest ...
+}
+```
+
+Then I add the type for 'target' to the globals as well:
+
+```ts
+const target: {
+  all?: Target;
+  [s: string]: Target;
+}
+interface Target {
+  (...args: any[]): void;
+  result?: any;
+  done?: boolean;
+}
+```
+
+This exposes a couple more errors. See the section on fixing errors in
+existing types for how to fix those.
+
+Now we want to publish this to Definitely Typed:
+
+1. Fork DefinitelyTyped on github.
+2. `git clone https://github.com/your-name-here/DefinitelyTyped`
+3. `cp node_modules/@types/shelljs/make.d.ts ~/DefinitelyTyped/types/shelljs/`
+4. `git checkout -b add-shelljs-make`
+5. Commit the change and push it to your github fork.
+6. Create a PR for the change.
+
+If there are lint problems, the CI run on Travis will catch them.
+
+For more detail on writing definitions for Definitely Typed, [see the
+Declaration section of the Typescript handbook](http://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html).
+
+## Fix references to types in dependencies
+
+In analyze-scope.js, I see quite a few errors about missing estree
+types like Identifier and ClassDeclaration. That's weird, because
```

Comment:
This confused me upon first read. Where are these types coming from? Aren't we in JS? I now understand they're coming from JSDoc. Maybe a source link would have helped unconfuse me faster.

Also use backticks ```